### PR TITLE
Add COIP Pools to EIP resource and datasource (Outpost)

### PR DESCRIPTION
**Pull request by johnbarney**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #12302

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
Add COIP Pools to EIP resource and datasource
```

Output from acceptance testing:

```
$ AWS_COIP_POOL_ID=ipv4pool-coip-<snip> make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAWSEIP_COIPPool'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccDataSourceAWSEIP_COIPPool -timeout 120m
=== RUN   TestAccDataSourceAWSEIP_COIPPool
=== PAUSE TestAccDataSourceAWSEIP_COIPPool
=== CONT  TestAccDataSourceAWSEIP_COIPPool
--- PASS: TestAccDataSourceAWSEIP_COIPPool (56.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	57.300s

$ AWS_COIP_POOL_ID=ipv4pool-coip-<snip> make testacc TEST=./aws TESTARGS='-run=TestAccAWSEIP_COIPPool'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEIP_COIPPool -timeout 120m
=== RUN   TestAccAWSEIP_COIPPool
=== PAUSE TestAccAWSEIP_COIPPool
=== CONT  TestAccAWSEIP_COIPPool
--- PASS: TestAccAWSEIP_COIPPool (49.60s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	50.674s
```

### CODE REVIEW

File: aws/data_source_aws_eip.go [hunk 1]
```diff
@@ -149,6 +157,18 @@ func dataSourceAwsEipRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	d.Set("public_ipv4_pool", eip.PublicIpv4Pool)
 
+	d.Set("public_ipv4_pool", eip.PublicIpv4Pool)
```
Comment (ewbankkit) [inline aws/data_source_aws_eip.go:160]: Already set above.

File: aws/data_source_aws_eip.go [hunk 2]
```diff
@@ -149,6 +157,18 @@ func dataSourceAwsEipRead(d *schema.ResourceData, meta interface{}) error {
...
+	if eip.CustomerOwnedIpv4Pool != nil {
```
Comment (ewbankkit) [inline aws/data_source_aws_eip.go:161]: No need for `nil` check here, `d.Set("customer_owned_ipv4_pool", eip.CustomerOwnedIpv4Pool)` is sufficient.

File: aws/resource_aws_eip.go [hunk 3]
```diff
@@ -257,6 +271,16 @@ func resourceAwsEipRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 	d.Set("public_ipv4_pool", address.PublicIpv4Pool)
+	if address.CustomerOwnedIpv4Pool != nil {
```
Comment (ewbankkit) [inline aws/resource_aws_eip.go:274]: No need for `nil` check here and below.

File: website/docs/d/eip.html.markdown [hunk 4]
```diff
@@ -75,6 +75,8 @@ In addition to all arguments above, the following attributes are exported:
 * `public_ip` - Public IP address of Elastic IP.
 * `public_dns` - Public DNS associated with the Elastic IP address.
 * `public_ipv4_pool` - The ID of an address pool.
+* `customer_owned_ipv4_pool` - The ID of a Customer Owned IP Pool.
```
Comment (ewbankkit) [inline website/docs/d/eip.html.markdown:78]: Maybe add a link - https://docs.aws.amazon.com/outposts/latest/userguide/outposts-networking-components.html#ip-addressing -?

File: website/docs/r/eip.html.markdown [hunk 5]
```diff
@@ -105,6 +105,7 @@ The following arguments are supported:
   the Elastic IP address is associated with the primary private IP address.
 * `tags` - (Optional) A mapping of tags to assign to the resource.
 * `public_ipv4_pool` - (Optional) EC2 IPv4 address pool identifier or `amazon`. This option is only available for VPC EIPs.
+* `customer_owned_ipv4_pool` - (Optional) The  ID  of a customer-owned address pool.
```
Comment (ewbankkit) [inline website/docs/r/eip.html.markdown:108]: Add a link as for `eip`?

**Comment by johnbarney** [comment]
```
$ AWS_COIP_POOL_ID=ipv4pool-coip-<snip> make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAWSEIP_COIPPool'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccDataSourceAWSEIP_COIPPool -timeout 120m
=== RUN   TestAccDataSourceAWSEIP_COIPPool
=== PAUSE TestAccDataSourceAWSEIP_COIPPool
=== CONT  TestAccDataSourceAWSEIP_COIPPool
--- PASS: TestAccDataSourceAWSEIP_COIPPool (66.58s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	69.038s
$ AWS_COIP_POOL_ID=ipv4pool-coip-<snip> make testacc TEST=./aws TESTARGS='-run=TestAccAWSEIP_COIPPool'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEIP_COIPPool -timeout 120m
=== RUN   TestAccAWSEIP_COIPPool
=== PAUSE TestAccAWSEIP_COIPPool
=== CONT  TestAccAWSEIP_COIPPool
--- PASS: TestAccAWSEIP_COIPPool (60.48s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	63.177s
```

**Pull request merged**